src/coreclr/jit/gentree.cpp

```diff
@@ -25802,9 +25802,9 @@ regNumber GenTree::ExtractTempReg(regMaskTP mask /* = (regMaskTP)-1 */)
 {
     regMaskTP availableSet = gtRsvdRegs & mask;
     assert(genCountBits(availableSet) >= 1);
-    regMaskTP tempRegMask = genFindLowestBit(availableSet);
-    gtRsvdRegs &= ~tempRegMask;
-    return genRegNumFromMask(tempRegMask);
+    regNumber tempReg = genFirstRegNumFromMask(availableSet);
+    gtRsvdRegs ^= genRegMask(tempReg);
```

Comment:
Is this actually faster than the previous code? Since it needs to do either a left shift (on amd64) or memory lookup (non-amd64). The same question applies to all the places where you introduced `genRegMask`.

It seems like you're saying `b = genRegMask(...)` + `a ^= b` is faster than `a &= ~b`?

The `genFirstRegNumFromMaskAndToggle` cases seem like a clear win, but I'm not as sure about these.

Reply:
`a ^= (1 << …)` is specially recognized and transformed into `btc` on xarch. There is sometimes special optimizations possible on Arm64 as well, but it’s worst case the same number of instructions and execution cost (but often slightly shorter)